engine/src/main/java/org/terasology/rendering/opengl/FrameBuffersManager.java

```diff
@@ -78,98 +89,171 @@
-     * to zero.
+     * Executed before any rendering begins, this method disposes and regenerates a number of FBOs if size changes
+     * have been triggered. Also prompts the GraphicState and PostProcessor instances to refresh their internal
+     * references to use the new FBOs.
      */
-    public void createOrUpdateFullscreenFbos() {
+    public void preRenderUpdate() {
+        refreshDynamicFBOsDimensions();
+        if (sceneOpaque.dimensions().areDifferentFrom(fullScale)) {
+            disposeOfAllDynamicFBOs();
+            recreateDynamicFBOs();
+            graphicState.refreshDynamicFBOs();
+            postProcessor.refreshDynamicFBOs();
+        }
 
-        if (overwriteRtWidth == 0) {
+        if (sceneShadowMap != null && sceneShadowMap.width() != renderingConfig.getShadowMapResolution()) {
+            recreateShadowMapFBO();
+            graphicState.setSceneShadowMap(sceneShadowMap);
+        }
+    }
+
+    private boolean refreshDynamicFBOsDimensions() {
+        boolean refreshed = false;
+        Dimensions oldFullScale = fullScale;
+
+        if (isNotTakingScreenshot()) {
             fullScale = new Dimensions(Display.getWidth(), Display.getHeight());
-        } else {
-            fullScale = new Dimensions(overwriteRtWidth, overwriteRtHeight);
             if (renderingConfig.isOculusVrSupport()) {
                 fullScale.multiplySelfBy(OculusVrHelper.getScaleFactor());
             }
+        } else {
+            fullScale = new Dimensions(
+                    renderingConfig.getScreenshotSize().getWidth(Display.getWidth()),
+                    renderingConfig.getScreenshotSize().getHeight(Display.getHeight())
+            );
         }
 
         fullScale.multiplySelfBy(renderingConfig.getFboScale() / 100f);
 
-        halfScale    = fullScale.dividedBy(2);   // quarter resolution
-        quarterScale = fullScale.dividedBy(4);   // one 16th resolution
-        one8thScale  = fullScale.dividedBy(8);   // one 64th resolution
-        one16thScale = fullScale.dividedBy(16);  // one 256th resolution
-        one32thScale = fullScale.dividedBy(32);  // one 1024th resolution
+        if (fullScale.isDifferentFrom(oldFullScale)) {
+            halfScale    = fullScale.dividedBy(2);
+            quarterScale = fullScale.dividedBy(4);
+            one8thScale  = fullScale.dividedBy(8);
+            one16thScale = fullScale.dividedBy(16);
+            one32thScale = fullScale.dividedBy(32);
+            refreshed = true;
+        }
 
-        FBO scene = fboLookup.get("sceneOpaque");
-        final boolean recreate = scene == null || (scene.dimensions().areDifferentFrom(fullScale));
+        return refreshed;
+    }
 
-        if (!recreate) {
-            return;
-        }
+    // providing a rough guide of each FBO here, as the method that creates them (recreateDynamicFBOs) is quite dense already
```

Comment:
Oh, right, what I just commented on above, heh. We *could* do little shell methods just to have a place for a nice description in javadoc. Some day.